doc/Architecture-overview/index.md

```diff
@@ -2,7 +2,158 @@
+
+A similar approach can be used for offloading all of the load from a node subject to a scheduled service, or to gradually migrate the cluster across the physical or virtual machines.
+Doing so requires setting of a custom load balancing strategy.
+
+# Endpoint SDK
+
+The Kaa endpoint SDK is a library which provides communication, data marshalling, persistence, and other functions available in Kaa for specific type of an endpoint (e.g. [C-based](Using-Kaa-C-endpoint-SDK), [C++-based](Using-Kaa-Cpp-endpoint-SDK), [Java-based](Using-Kaa-Java-endpoint-SDK), [Android-based](), [Objective-C-based](Using-Kaa-Objective-C-endpoint-SDK)).
+The client SDK abstracts the communication protocol, data persistence, and other implementation details that may be specific for any concrete solution based on Kaa.
+
+Endpoint SDK helps to save time on development routine and allows to concentrate on your business logic.
+
+
+# Kaa instance
+
+<img src="attach/logical-concepts.png" width="1382" height="703" style="display: block;margin-left: auto;margin-right: auto;">
+
+*Kaa instance* (interchangeable with *Kaa deployment*) is a particular installation of the Kaa platform, either as a [single node]({{root_url}}Administration-guide/System-installation/Single-node-installation/), or a [clustered deployment]({{root_url}}Administration-guide/System-installation/Cluster-setup/).
+
+An *application* in Kaa defines the set of data models, the corresponding flows among the endpoints and Kaa server, and processing rules.
+Kaa applications are agnostic to the specific target platform, operating system, or the client software implementation.
+For example, two firmware implementations for a pressure sensor will differ between Arduino and STM32 platforms, yet will be considered the same application in Kaa, as long as they report identically structured telemetry data.
+
+The Kaa platform is *multi-tenant*, and may support multiple independent business entities out of a single instance.
+As illustrated in the following diagram, applications belong to tenants, while endpoints register within applications.
+
+An *endpoint* (also commonly shortcut as *EP*), on its part, is an abstraction that represents a separate managed entity within a Kaa deployment.
+Practically speaking, an endpoint is a specific Kaa client registered (or waiting to be registered) within a Kaa instance.
+Depending on the use case, different level physical entities may be considered as endpoints.
+In the industrial setting, a single air quality sensor may represent an individual endpoint, while in a fleet tracking application, a truck (despite carrying on board multiple sensors reporting data) may be a more appropriate entity to be declared as an endpoint.
+
+In order to be able to distinguish endpoints not only by their *endpoint IDs*, but also associated properties, Kaa introduces a concept of [*endpoint profiles*]({{root_url}}Programming-guide/Key-platform-features/Endpoint-profiles/).
```

Comment:
It is a little bit difficult to understand. The "endpoint ID" was not introduced. Also, I am not sure that the ability to "distinguish endpoints not only by their *endpoint IDs*, but also associated properties" is the main feature of the endpoint profiles.